src/main/java/core/basesyntax/impl/FruitsFromFileImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.impl;
+
+import core.basesyntax.db.FruitShopDao;
+import core.basesyntax.service.FruitTransaction;
+import core.basesyntax.service.FruitsFromFile;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FruitsFromFileImpl implements FruitsFromFile {
+    private FruitShopDao fruitShopDao;
+
+    public FruitsFromFileImpl(FruitShopDao fruitShopDao) {
+        this.fruitShopDao = fruitShopDao;
+    }
+
+    @Override
+    public List<String> getFruits(List<FruitTransaction> fruitTransactions) {
+        List<String> fruits = fruitTransactions.stream()
+                .map(FruitTransaction::getFruit)
+                .distinct()
+                .collect(Collectors.toList());
+        for (String fruit : fruits) {
+            fruitShopDao.putOnlyFruits(fruit);
```

Comment:
Direct interaction with the DAO within a method that seems to be just a getter is not a good practice. It violates the Single Responsibility Principle as the method is doing more than just retrieving fruits. Consider refactoring to separate concerns.